processor/transformer/transformer.go

```diff
@@ -205,39 +206,29 @@ func (trans *HandleT) Transform(ctx context.Context, clientEvents []TransformerE
 
 	sTags := statsTags(clientEvents[0])
 
-	batchCount := len(clientEvents) / batchSize
-	if len(clientEvents)%batchSize != 0 {
-		batchCount += 1
-	}
+	batches := lo.Chunk(clientEvents, batchSize)
 
 	stats.Default.NewTaggedStat(
 		"processor.transformer_request_batch_count",
 		stats.HistogramType,
 		sTags,
-	).Observe(float64(batchCount))
-	trace.Logf(ctx, "request", "batch_count: %d", batchCount)
+	).Observe(float64(len(batches)))
+	trace.Logf(ctx, "request", "batch_count: %d", len(batches))
 
-	transformResponse := make([][]TransformerResponseT, batchCount)
+	transformResponse := make([][]TransformerResponseT, len(batches))
 
-	wg := sync.WaitGroup{}
-	wg.Add(len(transformResponse))
+	p := pool.New().WithMaxGoroutines(maxConcurrency)
 	for i := range transformResponse {
-		i := i
-		from := i * batchSize
-		to := (i + 1) * batchSize
-		if to > len(clientEvents) {
-			to = len(clientEvents)
-		}
-		trans.guardConcurrency <- struct{}{}
```

Comment:
But we still need to have a `HandleT`-wide concurrency guard, right? In the future, multiple go routines might be calling `HandleT#Transform`. If this doesn't happen already, it definitely will happen in the future when we will be implementing isolation at processor.

Reply:
you mean having this gaurd outside `Transform`..?